std/path.d

```diff
@@ -1053,6 +1052,47 @@ unittest
     assert (setExtension("file.ext", "") == "file");
 }
 
+/************
+ * Replace existing extension on filespec with new one.
+ *
+ * Params:
+ *      path = string or random access range representing a filespec
+ * Returns:
+ *      Range with $(D path)'s extension (if any) replaced with $(D ext).
+ *      The element encoding type of the returned range will be the same as $(D path)'s.
+ * See_Also:
+ *      $(LREF setExtension)
+ */
+auto setExt(R, C)(R path, C[] ext)
+    if ((isRandomAccessRange!R && hasSlicing!R && hasLength!R && isSomeChar!(ElementType!R) ||
+         is(StringTypeOf!R)) &&
+        isSomeChar!C)
+{
+    import std.range : only, chain;
+    import std.utf : byUTF;
+
+    alias CR = Unqual!(ElementEncodingType!R);
+    if (ext.length > 0 && ext[0] == '.')
+        ext = ext[1 .. $];              // remove any leading . from ext[]
+    auto dot = only(cast(CR)'.');
+    if (ext.length == 0)
+        dot.popFront();                 // so dot is an empty range, too
+    return chain(stripExtension(path).byUTF!CR, dot, ext.byUTF!CR);
```

Comment:
That isn't quite the same logic. If ext is empty, the . should not be included. Your rewrite includes it.